managed/models/agent_helpers.go

```diff
@@ -229,7 +229,9 @@ func FindAgents(q *reform.Querier, filters AgentFilters) ([]*Agent, error) {
 
 	agents := make([]*Agent, len(structs))
 	for i, s := range structs {
-		agents[i] = s.(*Agent) //nolint:forcetypeassert
+		agent := s.(*Agent) //nolint:forcetypeassert
+		DecryptAgent(agent)
```

Comment:
Shouldn't `DecryptAgent` return the same pointer to the agent? I'd much more convenient to use it as follows:

`agents[i] = DecryptAgent(s.(*Agent)) //nolint:forcetypeassert`

Reply:
returning the same pointer will make wrong assumption that initial object isn't changed